hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/SCMNodeManager.java

```diff
@@ -176,6 +178,7 @@ public SCMNodeManager(OzoneConfiguration conf,
     String dnLimit = conf.get(ScmConfigKeys.OZONE_DATANODE_PIPELINE_LIMIT);
     this.heavyNodeCriteria = dnLimit == null ? 0 : Integer.parseInt(dnLimit);
     this.scmContext = scmContext;
+    this.clock = clock;
```

Comment:
Where does `SCMNodeManager` use this clock?

Reply:
It is Used by TestSCMNodeManager ..Cant find its use in SCMNodeManager

Reply:
These are the places where the "last heartbeat" time is stored:

https://github.com/apache/ozone/blob/e4ff5fa1b9b6a8335e5e1d7a118c9c3d4b46281d/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/DatanodeInfo.java#L75

https://github.com/apache/ozone/blob/e4ff5fa1b9b6a8335e5e1d7a118c9c3d4b46281d/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/DatanodeInfo.java#L88-L90

They should use `clock.millis()` instead of `Time.monotonicNow()`.  We need to pass `clock` to `NodeStateManager` here:

https://github.com/apache/ozone/blob/e4ff5fa1b9b6a8335e5e1d7a118c9c3d4b46281d/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/SCMNodeManager.java#L151-L152

Reply:
There are other places where `Time.monotonicNow()` in `NodeStateManager`, `SCMNodeManager` and `DatanodeInfo`, I think we should replace those, too.

Reply:
Ack